UI/scripts/view.js

```diff
@@ -0,0 +1,90 @@
+
+const dateConfig = {
+  year: 'numeric',
+  month: 'numeric',
+  day: 'numeric',
+  hour: 'numeric',
+  minute: 'numeric',
+};
+
+class View {
+  constructor(model, isLoggedUser) {
+    this.photoPosts = model;
+    this.isLoggedUser = isLoggedUser;
+  }
+
+  addPost(post) {
+    if (this.photoPosts.addPhotoPost(post)) {
+      const wrapper = document.querySelector('#wrapper');
+      const newPost = View.getNodeWithHtml(post);
+      wrapper.insertAdjacentElement('afterbegin', newPost);
+      return true;
+    }
+    return false;
+  }
+
+  deletePost(id) {
+    if (this.photoPosts.removePhotoPost(id)) {
+      const childNode = document.querySelector(`[data-id="${id}"]`);
+      if (childNode) {
+        childNode.parentNode.removeChild(childNode);
+      }
+      return true;
+    }
+    return false;
+  }
+
+  editPost(id, edits) {
+    if (this.photoPosts.editPhotoPost(id, edits)) {
+      const childNode = document.querySelector(`[data-id="${id}"]`);
+      if (childNode) {
+        const editedChildNode = View.getNodeWithHtml(this.photoPosts.getPhotoPost(id));
+        childNode.parentNode.replaceChild(editedChildNode, childNode);
+      }
+      return true;
+    }
+    return false;
+  }
+
+  showPosts() {
+    this.photoPosts.getPhotoPosts().reverse().forEach((post) => {
+      const wrapper = document.querySelector('#wrapper');
```

Comment:
Critical: That what I'm talking about on lecture:
Do not search the same DOM element in cycle the same as any operations that can be done outside of cycle